spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/KubernetesDiscoveryClient.java

```diff
@@ -171,4 +178,9 @@ public int getOrder() {
 		return properties.order();
 	}
 
+	@Deprecated(forRemoval = true)
```

Comment:
this PR introduces kubernetes namespace provider support. For it, I need to use `Fabric8Utils::getApplicationNamespace`, which requires a `KubernetesNamespaceProvider`, which requires `Environment`. 

Since adding another variable to the constructor would introduce a breaking change, I am adding this one. I am currently updating an existing PR (that will only go in the next major release) where I drop this method and inject an Environment variable in the constructor.